pkg/operator/api/api.go

```diff
@@ -291,22 +299,24 @@ func (a *apiServer) ListResiliency(ctx context.Context, in *operatorv1pb.ListRes
 func (a *apiServer) ComponentUpdate(in *operatorv1pb.ComponentUpdateRequest, srv operatorv1pb.Operator_ComponentUpdateServer) error { //nolint:nosnakecase
 	log.Info("sidecar connected for component updates")
 	key := uuid.New().String()
```

Comment:
(Unrelated from your PR but can you change this to uuid.NewRandom() which returns an error rather than panic if there's not enough entropy?)